lib/NetPacksLib.cpp

```diff
@@ -594,7 +594,11 @@ DLL_LINKAGE void HeroRecruited::applyGs( CGameState *gs )
 
 	h->setOwner(player);
 	h->pos = tile;
-	h->movement =  h->maxMovePoints(true);
+	bool fresh = (static_cast<int>(h->movement) == -1);

```

Comment:
Well, -1 here is quite different from -1 elsewhere and might be confusing if more code is modified. That's why we use named enums and constants to make sure what the value ir refers too.
It might be nice, but not urgent, to use it for hero constructor as well. All in all, this code has been working for some time.